lib/exceptions.rb

```diff
@@ -0,0 +1,7 @@
+module ManageIQ
+  module API
+    module Common
+      class HeadersNotSet < ArgumentError; end
```

Comment:
This exceptions file name should also be namespaced, otherwise you run the risk of this colliding with other gems that have an exceptions at the top level... `require 'manageiq/api/common/exceptions'`